src/main/java/com/coveros/selenified/services/Check.java

```diff
@@ -0,0 +1,305 @@
+        } else {
+            getReporter().fail("", "Expected to find a response code of <b>" + expectedCode + "</b>", "Found a response code of <b>" + actualCode + "</b>");
+        }
+        return actualCode;
+    }
+
+    /**
+     * Checks the actual response json payload is equal to the expected
+     * response json payload, and writes that out to the output file
+     *
+     * @param expectedJson - the expected response json object
+     */
+    void equals(JsonObject expectedJson);
+
+    /**
+     * Checks the actual response json payload is equal to the expected
+     * response json payload, and writes that out to the output file
+     *
+     * @param expectedJson - the expected response json object
+     */
+    default JsonObject checkEquals(JsonObject expectedJson) {
+        boolean pass = false;
+        JsonObject actualJson = null;
+        if (getResponse().getObjectData() != null) {
+            actualJson = getResponse().getObjectData();
+            pass = actualJson.equals(expectedJson);
+        }
+        if (pass) {
+            getReporter().pass("", EXPECTED_TO_FIND + DIV_I + Reporter.formatHTML(gson.toJson(expectedJson)) + END_IDIV, FOUND + Reporter.formatResponse(getResponse()));
+        } else {
+            getReporter().fail("", EXPECTED_TO_FIND + DIV_I + Reporter.formatHTML(gson.toJson(expectedJson)) + END_IDIV, FOUND + Reporter.formatResponse(getResponse()));
+        }
+        return actualJson;
+    }
+
+    /**
+     * Checks the actual response json payload is equal to the expected
+     * response json payload, and writes that out to the output file
+     *
+     * @param expectedJson - the expected response json array
+     */
+    void equals(JsonArray expectedJson);
+
+    /**
+     * Checks the actual response json payload is equal to the expected
+     * response json payload, and writes that out to the output file
+     *
+     * @param expectedJson - the expected response json array
+     */
+    default JsonArray checkEquals(JsonArray expectedJson) {
+        boolean pass = false;
+        JsonArray actualJson = null;
+        if (getResponse().getArrayData() != null) {
+            actualJson = getResponse().getArrayData();
+            pass = actualJson.equals(expectedJson);
+        }
+        if (pass) {
+            getReporter().pass("", EXPECTED_TO_FIND + DIV_I + Reporter.formatHTML(gson.toJson(expectedJson)) + END_IDIV, FOUND + Reporter.formatResponse(getResponse()));
+        } else {
+            getReporter().fail("", EXPECTED_TO_FIND + DIV_I + Reporter.formatHTML(gson.toJson(expectedJson)) + END_IDIV, FOUND + Reporter.formatResponse(getResponse()));
+        }
+        return actualJson;
+    }
+
+    /**
+     * Checks the actual response payload is equal to the expected
+     * response payload, and writes that out to the output file
+     *
+     * @param expectedMessage - the expected response message
+     */
+    void equals(String expectedMessage);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either override Object.equals(Object), or rename the method to prevent any confusion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1201)